py_gtfs_rt_ingestion/lib/convert_gtfs_rt.py

```diff
@@ -55,65 +58,69 @@ def __init__(self, config_type: ConfigType) -> None:
         now = datetime.now(tz=timezone.utc)
         self.start_of_hour = now.replace(minute=0, second=0, microsecond=0)
 
-        self.current_time: Optional[datetime] = None
-        self.table = pyarrow.table(
-            self.detail.empty_table(), schema=self.detail.export_schema
-        )
-        self.next_table: Optional[pyarrow.table] = None
-
-    def get_tables(self) -> List[Tuple[str, pyarrow.Table]]:
-        return [(str(self.config_type), self.table)]
+        self.error_files: List[str] = []
+        self.archive_files: List[str] = []
 
-    def reset(self) -> None:
-        self.current_time = None
-        self.table = self.next_table
-        self.next_table = None
+    def convert(self) -> None:
+        process_logger = ProcessLogger("gtfs_rt_convert")
+        process_logger.log_start()
 
-        self.archive_files = []
-        self.error_files = []
+        for table in self.process_files():
+            self.write_and_archive(table)
+            process_logger.log_complete()
+            process_logger.log_start()
 
-    def add_file(self, file: str) -> bool:
-        process_logger = ProcessLogger(
-            "convert_single_gtfs_rt",
-            config_type=str(self.config_type),
-            filename=file,
+    def process_files(self) -> Iterable[pyarrow.table]:
```

Comment:
this function looks a lot like what we had before, which makes sense since each lambda was essentially a parallelized unit.